### ISSUE TITLE
Caffe Regression: net prediction is much larger than label but the training loss is very small

### ISSUE THREAD

**Issue by YoungsonZhao** [BVLC/caffe#4936]
I am doing regression using caffe right now and I have met a weird problem. The net prediction is much larger than the label, however the training loss is very small. For example, the label is 7 and the prediction is 1451, the label is 34 and the prediction is 7038, the label is 52 and the prediction is 12238. (Both the label and the net prediction are float, there are listed an int for convenience here.)

I have made sure that the data and label in LMDB is correct. It seems that the label is scaled when loaded for training. Otherwise, the loss would be large (label is 34 and prediction is 7038). 

The data I used is 64*48 images, such as
1.jpg 25
2.jpg 7
3.jpg 34
4.jpg 58

The net is defined as
name: "zhnet_regression"

layer {
  name: "data"
  type: "Data"
  top: "image"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    scale: 0.00390625
  }
  data_param {
    source: "/home/yongshengzhao/Documents/CaffeProjects/SkyEye/data/train_regression_image_lmdb"
    backend: LMDB
    batch_size: 32
  }
}

#layer {
#  name: "label"
#  type: "Data"
#  top: "label"
#  include {
#    phase: TRAIN
#  }
#  transform_param {
#    scale: 1
#  }
#  data_param {
#    source: "/home/yongshengzhao/Documents/CaffeProjects/SkyEye/data/train_regression_label_lmdb"
#    backend: LMDB
#    batch_size: 32
#  }
#}

layer {
  name: "data"
  type: "Data"
  top: "image"
  top: "label"
  include {
    phase: TEST
  }
  transform_param {
    scale: 0.00390625
  }
  data_param {
    source: "/home/yongshengzhao/Documents/CaffeProjects/SkyEye/data/test_regression_image_lmdb"
    backend: LMDB
    batch_size: 32
  }
}

#layer {
#  name: "label"
#  type: "Data"
#  top: "label"
#  include {
#    phase: TEST
#  }
#  transform_param {
#    scale: 1
#  }
#  data_param {
#    source: "/home/yongshengzhao/Documents/CaffeProjects/SkyEye/data/test_regression_label_lmdb"
#    backend: LMDB
#    batch_size: 32
#  }
#}

layer {
  name: "conv1"
  type: "Convolution"
  bottom: "image"
  top: "conv1"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  convolution_param {
    num_output: 32
    kernel_size: 3
    pad: 0
    stride: 1
  }
}

layer {
  name: "ReLU1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}

layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool:MAX
    kernel_size: 2
    stride: 1
  }
}

layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  convolution_param {
    num_output: 8
    kernel_size: 3
    pad: 0
    stride: 1
  }
}

layer {
  name: "ReLU2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}

layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}

layer {
  name: "ip_1"
  type: "InnerProduct"
  bottom: "pool2"
  top: "ip1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 192
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "ip1"
  top: "ip1"
}

layer {
  name: "ip_2"
  type: "InnerProduct"
  bottom: "ip1"
  top: "ip2"
  inner_product_param {
    num_output: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
    type: "constant"
    value: 0
    }
  }
}

layer {
  name: "loss"
  type: "EuclideanLoss"
  bottom: "label"
  bottom: "ip2"
  top: "loss"
}

The solver is defines as:

# The train/test net protocol buffer definition
net: "/home/yongshengzhao/Documents/CaffeProjects/SkyEye/zhnet_regression.prototxt"
# test_iter specifies how many forward passes the test should carry out.
# In the case of MNIST, we have test batch size 100 and 100 test iterations,
test_initialization: true
# covering the full 10,000 testing images.
test_iter: 30
# Carry out testing every 500 training iterations.
test_interval: 50
# The base learning rate, momentum and the weight decay of the network.
base_lr: 0.0001
momentum: 0.9
weight_decay: 0.00025
# The learning rate policy
lr_policy: "inv"
gamma: 0.0001
power: 0.75
# Display every 100 iterations
display: 50
# The maximum number of iterations
max_iter: 1000
# snapshot intermediate results
snapshot: 1000
snapshot_prefix: "/home/yongshengzhao/Documents/CaffeProjects/SkyEye/zhnet_regression"
# solver mode: CPU or GPU
solver_mode: CPU

The deployed net is 
name: "zhnet_regression"

input: "image"
input_shape {
  dim: 1
  dim: 3
  dim: 48
  dim: 64
}

layer {
  name: "conv1"
  type: "Convolution"
  bottom: "image"
  top: "conv1"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  convolution_param {
    num_output: 32
    kernel_size: 3
    pad: 0
    stride: 1
  }
}

layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}

layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool:MAX
    kernel_size: 2
    stride: 1
  }
}

layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 0
  }
  param {
    lr_mult: 0
  }
  convolution_param {
    num_output: 8
    kernel_size: 3
    pad: 0
    stride: 1
  }
}

layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}

layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}

layer {
  name: "ip_1"
  type: "InnerProduct"
  bottom: "pool2"
  top: "ip1"
  param {
    lr_mult: 1
  }
  param {
    lr_mult: 2
  }
  inner_product_param {
    num_output: 192
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "ip1"
  top: "ip1"
}

layer {
  name: "ip_2"
  type: "InnerProduct"
  bottom: "ip1"
  top: "ip2"
  inner_product_param {
    num_output: 1
  }
}

The result looks weird to me. If the prediction is 7038 and the actual label is 34, the loss would be very large, but it is small in training. I have checked the LMDB file and made sure that the data and label is correct. What' the problem? I wonder that dose caffe automatically scale the label for regression? Or is there any thing wrong for Euclideanloss?

**Comment by williford** [comment]
Questions like these should be asked on the [Caffe user mailing list](https://groups.google.com/forum/#!forum/caffe-users) or on another website, such as http://stackoverflow.com/ or http://stats.stackexchange.com/. Please **close this issue** and move your question to one of these places.

**Comment by c437yuyang** [comment]
Have you solved this problem? I have the same weird problem

**Comment by shelhamer** [comment]
From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
